TW changes, consolidated files, and fixed broken links

jcayx opened the pull request:
- made several changes as requested by TW
- consolidated + removed /docs/howto/platform-sdk-quickstart-guide.md and /docs/references/platform-sdk.md
- fixed any broken links resulting from the file consolidation

jcayx committed:
TW changes, consolidated files, and fixed broken links

paul-england reviewed (commented):
Please make getting_started_markdown.md and `differences_with_original_sdk.md` kebab case and not snake case?

paul-england commented on docs/howto/getting_started_markdown.md:
```diff
@@ -146,6 +146,180 @@ development. For more information on Configuration Panel GUI
+
+The steps below illustrate how to use the latest developer tools to
+create a custom plugin or tool for use in Alteryx. For more information
+about new SDKs and their usage within Alteryx, please visit [Platform
```
No newline between Platform and SDK

jcayx replied:
removed

paul-england commented on docs/howto/getting_started_markdown.md:
```diff
@@ -146,6 +146,180 @@ development. For more information on Configuration Panel GUI
+Before you get started, make sure that these items are installed on your
+machine:
+
+-   Microsoft Windows 7 or Later (64-bit)
```
"later" should be lower case.

jcayx replied:
fixed

paul-england commented on docs/howto/getting_started_markdown.md:
```diff
@@ -146,6 +146,180 @@ development. For more information on Configuration Panel GUI
+machine:
+
+-   Microsoft Windows 7 or Later (64-bit)
+-   Alteryx Designer Version 2021.2\*
```
Why the asterisk?

jcayx replied:
here it indicates the note below the list with the asterisk is related to this line. Bit confusing imo but this information matches our other documentation

paul-england commented on docs/howto/getting_started_markdown.md:
```diff
@@ -146,6 +146,180 @@ development. For more information on Configuration Panel GUI
+-   [node](https://nodejs.org/en/download/) [14](https://nodejs.org/en/blog/release/v14.17.3/)
+-   [Git](https://git-scm.com/downloads)
+
+\*Alteryx Designer version 2021.4 or greater is required to use the Alteryx Python
```
Why the asterisk?

jcayx replied:
see above comment

paul-england commented on docs/howto/getting_started_markdown.md:
```diff
@@ -146,6 +146,180 @@ development. For more information on Configuration Panel GUI
+
+1.  Run this command to install the AYX Plugin CLI:\
+    \
+    `pip install ayx-plugin-cli`\
```
These are already covered in Install at the top of the doc.

jcayx replied:
removed this section and included link to the earlier part of the doc

jcayx committed:
renamed files, typographical changes from reviews

jcayx commented:
> Please make getting_started_markdown.md and `differences_with_original_sdk.md` kebab case and not snake case?

changed all markdown and image files to kebab case, fixed any links using the old filenames too. Left the python files as snake case since that seems to be our standard
